app/client/components/holiday/collatedCredits.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { friendlyLongDateFormat, HolidayStopDetail } from "./holidayStopApi";
+
+interface CollatedCreditByInvoiceDate {
+  [invoiceDateString: string]: number;
+}
+
+const collateCreditsByInvoiceDate = (
+  publicationsImpacted: HolidayStopDetail[]
+) =>
+  publicationsImpacted.reduce(
+    (accumulator, currentValue) => {
+      const credit = currentValue.actualPrice || currentValue.estimatedPrice;
+      if (currentValue.invoiceDate && credit) {
+        const invoiceDateAsString = currentValue.invoiceDate.format(
+          friendlyLongDateFormat
+        );
+        return {
+          ...accumulator,
+          [invoiceDateAsString]:
+            credit + (accumulator[invoiceDateAsString] || 0)
+        };
+      }
+      return {}; // if we don't have credits and dates for every entry we shouldn't show anything
```

Comment:
in talking this through with @frj this is potentially about flawed as it will could continue to accumulate after the missing entry - I'll figure out a better implementation